components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeService.java

```diff
@@ -193,19 +181,11 @@ public boolean isScopeExists(String name) throws IdentityOAuth2ScopeException {
      */
     public void deleteScope(String name) throws IdentityOAuth2ScopeException {
 
-        int tenantID = Oauth2ScopeUtils.getTenantID();
-        if (name == null) {
-            throw Oauth2ScopeUtils.generateClientException(Oauth2ScopeConstants.ErrorMessages.
-                    ERROR_CODE_BAD_REQUEST_SCOPE_NAME_NOT_SPECIFIED, null);
-        }
-
-        // check whether a scope exists with the provided scope name which to be deleted
-        boolean isScopeExists = isScopeExists(name);
-        if (!isScopeExists) {
-            throw Oauth2ScopeUtils.generateClientException(Oauth2ScopeConstants.ErrorMessages.
-                    ERROR_CODE_NOT_FOUND_SCOPE, name);
-        }
+        validateScopeName(name);
```

Comment:
The same methods are implemented in OAuthAdminServiceImpl as well. Can we get rid of one implementation and invoke the other service's method.

Reply:
Even though the names of the methods are same but internal logic is different, two different OSGI services hence keep as it is with two methods.